dso-l1/src/main/java/com/tc/object/InFlightMessage.java

```diff
@@ -336,23 +336,31 @@ public synchronized void setResult(byte[] value, EntityException error) {
       this.pendingAcks.clear();
       this.exception = error;
       this.getCanComplete = true;
+      handleException(error);// TODO: why isn't handleException called somewhere else?
```

Comment:
This will not work for the synchronous mode.  In synchronous mode, the monitor does not receive the result as that is returned by delivery of the future.

Reply:
I think I see what you mean: `InFlightMessage` should discriminate between sync and async mode and only call the two `handle*` methods in the async case, and `notifyAll` in the sync case. Let me try this.

Reply:
I've handled this case by discriminating between sync and async when creating the `InFlightMessage`. Hacky, but didn't require much engineering.